firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InAppMessageStreamManager.java

```diff
@@ -232,22 +243,37 @@ private boolean shouldIgnoreCache(String event) {
                       .defaultIfEmpty(CampaignImpressionList.getDefaultInstance())
                       .onErrorResumeNext(Maybe.just(CampaignImpressionList.getDefaultInstance()));
 
+              Maybe<InstanceIdResult> getIID = taskToMaybe(firebaseInstanceId.getInstanceId());
```

Comment:
Previously: When IID was failing, you were previously returning an expired response from the API Client.
I presume this would be cached, and the next time you lookup the cache, you would find that it has expired and make a service call.

Now: You this maybe would resolve to an error, that on line 274 would print the service failure.

Can you confirm that this logic is accurate by actually testing IID Fetch failure ?

Reply:
So right now the behavior is essentially FIAM will fetch infinitely if it encounters an error